docs/code/introduction/mode_code.md

```diff
@@ -0,0 +1,126 @@
+---
+title: "How to add custom Python code to a game mode"
+---
+
+
+# How to add custom Python code to a game mode
+
+The easiest and most common way to add custom Python code into your MPF game is to add a code module to a mode folder. That lets you run code when that mode is active and helps you break up any custom code you write per mode.
+
+This “mode code” (as we call it) has access to the full MPF API. You can post events, register event handlers which run custom things when events are posted, access device state and control devices, read and set player variables, post slides… really anything MPF can do, you can do.
+
+Here’s how you get started with custom mode code.
+
+## Create the module (file) to hold you code
+
+First, go into the mode folder where you want to create your custom code, and add a “code” folder to that mode folder. Then inside that folder, create a file (we usually give this file the same name as the mode) with a .py extension.
```

Comment:
It would be nice to note that your mode folder should already have a "config" folder in it that holds the <mode_name>.yaml file, and the "code" folder goes next to the "config" folder.

Reply:
made a directory tree to show this, should be clearer now.